src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -181,36 +128,73 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
                         Reporter.WriteLine(string.Format(LocalizableStrings.WorkloadAlreadyInstalled, string.Join(" ", previouslyInstalledWorkloads)).Yellow());
                     }
                     workloadIds = workloadIds.Concat(installedWorkloads).Distinct();
+                    workloadIds = WriteSDKInstallRecordsForVSWorkloads(workloadIds);
 
-                    var useWorkloadSets = ShouldUseWorkloadSetMode(_sdkFeatureBand, _dotnetPath);
-                    useRollback = !string.IsNullOrWhiteSpace(_fromRollbackDefinition);
-
-                    _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(includePreviews, useWorkloadSets, offlineCache).Wait();
-
-                    if (useWorkloadSets)
+                    if (!_skipManifestUpdate)
                     {
-                        manifestsToUpdate = InstallWorkloadSet(context);
+                        var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
+                        if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) && File.Exists(installStateFilePath) && InstallStateContents.FromString(File.ReadAllText(installStateFilePath)).Manifests is not null)
```

Comment:
```suggestion
                        if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) && InstallStateContents.FromPath(installStateFilePath)?.Manifests is not null)
```

This is what FromPath is for 🙂